### PR TITLE
Feature(profile): 15380 - integrate logto into profile

### PR DESCRIPTION
**Pull request by SamSalvatico**
### Ticket:

- [15380](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/15380)

### Description

Started integrating Profile with Logto

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**

### CODE REVIEW

File: apps/profile-api/package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "@fastify/type-provider-typebox": "^4.0.0",
     "@fastify/under-pressure": "^8.3.0",
     "@sinclair/typebox": "^0.32.16",
+    "api-auth": "*",
```
Comment (alfonsograziano) [inline apps/profile-api/package.json:41]: non-blocking: do you want to add the api-auth in the nodemon watch config as well? I guess is just a nice to have :)

File: apps/profile/.env.sample [hunk 2]
```diff
@@ -11,5 +11,9 @@ HOST_URL=http://localhost:3003/
...
 MESSAGES_HOST_URL=http://localhost:3002/
-
-PROFILE_BACKEND_URL=http://localhost:8003/
\ No newline at end of file
+NEXT_PUBLIC_PROFILE_SERVICE_ENTRY_POINT=http://localhost:3003
+PROFILE_BACKEND_URL=http://localhost:8003/
+LOGTO_PROFILE_APP_ID=RZmVtZUnrv01YzUYEIwYW
+LOGTO_PROFILE_APP_SECRET=profile_app_local_secret
+LOGTO_COOKIE_SECRET=profile_app_cookie_secret
+LOGTO_ENDPOINT=http://localhost:3301/
```
Comment (alfonsograziano) [inline apps/profile/.env.sample:19]: EOF please :)

**Review by edge33 — APPROVED** [review]
LGTM

**Review by nnorbert — APPROVED** [review]
LGTM!

**Pull request merged**